### PR TITLE
backquote column names in `default` and `generated` column expressions

### PR DESCRIPTION
**Pull request by jycor**
To prevent parsing errors for special column names, we need to wrap column identifiers with backquotes.
MySQL prints column identifiers in `default` and `generated` expressions when doing `show create table ...` with backquotes anyways.

fixes https://github.com/dolthub/dolt/issues/7388

### CODE REVIEW

File: memory/database.go [hunk 1]
```diff
@@ -227,14 +228,41 @@ func (d *BaseDatabase) AddTable(name string, t MemTable) {
 	d.tables[name] = t
 }
 
+func BacktickDefaultColumnValueNames(sch sql.Schema) sql.Schema {
```
Comment (max-hoffman) [inline memory/database.go:231]: is there a dependency cycle if this is in planbuilder?

Comment (zachmu) [inline memory/database.go:231]: Why is this here? Put this in the rowexec package?

**Review by zachmu — APPROVED** [review]
Good tests and solution, but question about the code placement

File: memory/database.go (hunk 1 above)
Comment (jycor) [reply memory/database.go:231]: yes

Comment (jycor) [reply memory/database.go:231]: putting this in the rowexec package results in an import cycle

**Comment by jycor** [comment]
closing for better solution here:
https://github.com/dolthub/go-mysql-server/pull/2291
